modules/blocks/utils.js

```diff
@@ -332,18 +336,119 @@ Utils.prototype.loadBlocksData = function(filter, cb, tx) {
+ *
+ * @param {Object} ExtendedBlock - Storage ExtendedBlock object
+ * @returns {Array} Array of transactions with block data formated as full_blocks_list db view
+ */
+Utils.prototype._parseStorageObjToLegacyObj = function(block) {
+	const parsedBlocks = [];
+	let transactions = [{}];
+
+	if (Array.isArray(block.transactions) && block.transactions.length > 0) {
+		transactions = block.transactions;
+	}
+
+	/* eslint-disable no-restricted-globals */
+	transactions.forEach(t => {
+		parsedBlocks.push({
+			b_id: _.get(block, 'id', null),
+			b_version: isNaN(+block.version) ? null : +block.version,
+			b_timestamp: isNaN(+block.timestamp) ? null : +block.timestamp,
+			b_height: isNaN(+block.height) ? null : +block.height,
+			b_previousBlock: _.get(block, 'previousBlockId', null),
+			b_numberOfTransactions: isNaN(+block.numberOfTransactions)
+				? null
+				: +block.numberOfTransactions,
+			b_totalAmount: _.get(block, 'totalAmount', null),
+			b_totalFee: _.get(block, 'totalFee', null),
+			b_reward: _.get(block, 'reward', null),
+			b_payloadLength: isNaN(+block.payloadLength)
+				? null
+				: +block.payloadLength,
+			b_payloadHash: _.get(block, 'payloadHash', null),
+			b_generatorPublicKey: _.get(block, 'generatorPublicKey', null),
+			b_blockSignature: _.get(block, 'blockSignature', null),
+			t_id: _.get(t, 'id', null),
+			t_type: _.get(t, 'type', null),
+			t_timestamp: _.get(t, 'timestamp', null),
+			t_senderPublicKey: _.get(t, 'senderPublicKey', null),
+			t_senderId: _.get(t, 'senderId', null),
+			t_recipientId: _.get(t, 'recipientId', null),
+			t_amount: _.get(t, 'amount', null),
+			t_fee: _.get(t, 'fee', null),
+			t_signature: _.get(t, 'signature', null),
+			t_signSignature: _.get(t, 'signSignature', null),
+			t_requesterPublicKey: _.get(t, 'requesterPublicKey', null),
+			t_signatures: t.signatures ? t.signatures.join(',') : null,
+			tf_data: _.get(t, 'asset.data', null),
+			s_publicKey: _.get(t, 'asset.signature.publicKey', null),
+			d_username: _.get(t, 'asset.delegate.username', null),
+			v_votes: t.asset && t.asset.votes ? t.asset.votes.join(',') : null,
+			m_min: _.get(t, 'asset.multisignature.min', null),
+			m_lifetime: _.get(t, 'asset.multisignature.lifetime', null),
+			m_keysgroup: _.get(t, 'asset.multisignature.keysgroup', null),
+			dapp_name: _.get(t, 'asset.dapp.name', null),
+			dapp_description: _.get(t, 'asset.dapp.description', null),
+			dapp_tags: _.get(t, 'asset.dapp.tags', null),
+			dapp_type: _.get(t, 'asset.dapp.type', null),
+			dapp_link: _.get(t, 'asset.dapp.link', null),
+			dapp_category: _.get(t, 'asset.dapp.category', null),
+			dapp_icon: _.get(t, 'asset.dapp.icon', null),
+			in_dappId: _.get(t, 'asset.inTransfer.dappId', null),
+			ot_dappId: _.get(t, 'asset.outTransfer.dappId', null),
+			ot_outTransactionId: _.get(t, 'asset.outTransfer.transactionId', null),
+		});
+	});
+
+	return parsedBlocks;
+};
```

Comment:
I would suggest to move this whole logic outside the file and implement thought decorator pattern. 

https://www.dofactory.com/javascript/decorator-design-pattern

This will help you to decorate the block in formats supported by specific versions.

Reply:
I like the idea of using decorators but in this case I don't think it's worth the effort because this whole function will be deleted once we fix this undesired behaviour.

Reply:
Ok Agreed.